templates/rhdm70-optashift-externaldb.yaml

```diff
@@ -0,0 +1,193 @@
+  name: IMAGE_STREAM_NAMESPACE
+  value: openshift
+  required: true
+- displayName: ImageStream Tag
+  description: A named pointer to an image in an image stream. Default is "1.0".
+  name: IMAGE_STREAM_TAG
+  value: "1.0"
+  required: false
+- displayName: "Datasource Minimum Pool Size"
+  description: "Sets xa-pool/min-pool-size for the configured datasource."
+  name: DB_MIN_POOL_SIZE
+  required: false
+- displayName: "Datasource Maximum Pool Size"
+  description: "Sets xa-pool/max-pool-size for the configured datasource."
+  name: DB_MAX_POOL_SIZE
+  required: false
+- displayName: "Datasource Transaction Isolation"
+  description: "Sets transaction-isolation for the configured datasource."
+  name: DB_TX_ISOLATION
+  required: false
+- displayName: Optashift rostering persistence Dialect
+  description: Optashift rostering server persistence dialect 
+  name: OPTASHIFT_ROSTERING_PERSISTENCE_DIALECT
+  value: ''
+  required: false
+- displayName: Optashift rostering persistence Datasource
+  description: Optashift rostering server persistence datasource for persistence.xml 
+  name: OPTASHIFT_ROSTERING_PERSISTENCE_DATASOURCE
+  value: ''
+  required: false
+- description: Database name
+  displayName: Database Name
+  name: DBE_DATABASE
+  required: true
+- description: Database user name
+  displayName: Database Username
+  name: DBE_USERNAME
+  required: true
+- description: Database user password
+  displayName: Database Password
+  name: DBE_PASSWORD
+  required: true
+- description: Database driver, e.g. postgresql, mysql
+  displayName: Database driver
+  name: DBE_DRIVER
+  required: true
+  value: mysql  
+- description: Database service host name
+  displayName: Database service host
+  name: DBE_SERVICE_HOST
+  required: true
+- description: Database service port number
+  displayName: Database service port
+  name: DBE_SERVICE_PORT
+  required: true
+- description: This parameter allows to configure datasource or xa-datasource
+  displayName: Database datasource / xa-datasource
+  name: NON_XA_DATASOURCE
+  value: "true"
+objects:
+- kind: Service
+  apiVersion: v1
+  spec:
+    ports:
+    - port: 8080
+      targetPort: 8080
+    selector:
+      deploymentConfig: "${APPLICATION_NAME}-optashift"
+  metadata:
+    name: "${APPLICATION_NAME}-optashift"
+    labels:
+      application: "${APPLICATION_NAME}"
+    annotations:
+      description: The execution server web server's http port.
+- kind: Route
+  apiVersion: v1
+  id: "${APPLICATION_NAME}-optashift-http"
+  metadata:
+    name: "${APPLICATION_NAME}-optashift"
+    labels:
+      application: "${APPLICATION_NAME}"
+    annotations:
+      description: Route for execution server's http service.
+  spec:
+    host: "${EXECUTION_SERVER_HOSTNAME_HTTP}"
+    to:
+      name: "${APPLICATION_NAME}-optashift"
+- kind: ServiceAccount
+  apiVersion: v1
+  metadata:
+    name: optashift-service-account
```

Comment:
What is the reason to use service account here? Does OptaShift use any OpenShift API in its scripts?